src/app/clusters/webrtc-transport-provider-server/webrtc-transport-provider-server.cpp

```diff
@@ -0,0 +1,495 @@
+
+    if (mDelegate == nullptr)
+    {
+        ChipLogError(Zcl, "No delegate set for WebRTCTransportProviderServer");
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Failure);
+        return;
+    }
+
+    switch (ctx.mRequestPath.mCommandId)
+    {
+    case Commands::SolicitOffer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::SolicitOffer::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleSolicitOffer(subCtx, req); });
+        break;
+
+    case Commands::ProvideOffer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvideOffer::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideOffer(subCtx, req); });
+        break;
+
+    case Commands::ProvideAnswer::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvideAnswer::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideAnswer(subCtx, req); });
+        break;
+
+    case Commands::ProvideICECandidates::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ProvideICECandidates::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleProvideICECandidates(subCtx, req); });
+        break;
+
+    case Commands::EndSession::Id:
+        CommandHandlerInterface::HandleCommand<Commands::EndSession::DecodableType>(
+            ctx, [this](HandlerContext & subCtx, const auto & req) { HandleEndSession(subCtx, req); });
+        break;
+
+    default:
+        // Mark unrecognized command as UnsupportedCommand
+        ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Status::UnsupportedCommand);
+        break;
+    }
+}
+
+// Helper functions
+WebRTCSessionStruct * WebRTCTransportProviderServer::FindSession(uint16_t sessionId)
+{
+    for (auto & session : mCurrentSessions)
+    {
+        if (session.id == sessionId)
+        {
+            return &session;
+        }
+    }
+
+    return nullptr;
+}
+
+void WebRTCTransportProviderServer::AddOrUpdateSession(const WebRTCSessionStruct & session)
+{
+    bool found = false;
+
+    // Search for a session with the same ID and update if found
+    for (auto & existing : mCurrentSessions)
+    {
+        if (existing.id == session.id)
+        {
+            existing = session;
+            found    = true;
+            break;
+        }
+    }
+
+    // If not found, add a new entry
+    if (!found)
+    {
+        mCurrentSessions.push_back(session);
+    }
+
+    // Notify the stack that the CurrentSessions attribute has changed
+    MatterReportingAttributeChangeCallback(AttributeAccessInterface::GetEndpointId().Value(), WebRTCTransportProvider::Id,
+                                           WebRTCTransportProvider::Attributes::CurrentSessions::Id);
+}
+
+void WebRTCTransportProviderServer::RemoveSession(uint16_t sessionId)
+{
+    size_t originalSize = mCurrentSessions.size();
```

Comment:
https://en.cppreference.com/w/cpp/container/vector/erase2 says that the return value is the number of erased elements. We could use that instead of size checks as well.

Reply:
This is erase_if, but it is feature of C++20, we don't support

```
    size_t erasedCount = std::erase_if(
        mCurrentSessions, 
        [sessionId](const WebRTCSessionStruct & s) {
            return s.id == sessionId;
        }
    );

    if (erasedCount > 0)
    {
        MatterReportingAttributeChangeCallback(
            AttributeAccessInterface::GetEndpointId().Value(),
            WebRTCTransportProvider::Id,
            WebRTCTransportProvider::Attributes::CurrentSessions::Id
        );
    }
```